### PR TITLE
Mssql destination: enable DAT tests, use nvarchar and datetime2 by default

### PR DESCRIPTION
**Pull request by sashaNeshcheret**
## What
Enabled DAT tests for Mssql destination, updated data types.

## How
Overridden methods for enabling DAT tests for Mssql destination, replace varchar to nvarchar for supporting special characters, and datetime to datetime2 for increasing range of values.

## Recommended reading order
1. `datatypes.sql`
2. `MSSQLDestinationAcceptanceTest.java`
3. `MSSQLTestDataComparator.java`
4. `x.java`

## 🚨 User Impact 🚨
Are there any breaking changes? What is the end result perceived by the user? If yes, please merge this PR with the 🚨🚨 emoji so changelog authors can further highlight this if needed.

## Pre-merge Checklist
Expand the relevant checklist and delete the others.

<details><summary><strong>New Connector</strong></summary>

### Community member or Airbyter

- [ ] **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] `docs/SUMMARY.md`
    - [ ] `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [ ] `docs/integrations/README.md`
    - [ ] `airbyte-integrations/builds.md`
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the connector is published, connector added to connector index as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)

</details>

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- [x] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [x] Secrets in the connector's spec are annotated with `airbyte_secret`
- [x] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub and connector version bumped by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)

</details>

<details><summary><strong>Connector Generator</strong></summary>

- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- [ ] The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:testScaffoldTemplates` then checking in your changes
- [ ] Documentation which references the generator is updated as needed

</details>

## Tests

<details><summary><strong>Unit</strong></summary>

*Put your unit tests output here.*

</details>

<details><summary><strong>Integration</strong></summary>

*Put your integration tests output here.*

</details>

<details><summary><strong>Acceptance</strong></summary>

*Put your acceptance tests output here.*

</details>

### CODE REVIEW

File: airbyte-integrations/bases/base-normalization/integration_tests/resources/test_simple_streams/data_input/messages.txt [hunk 1]
```diff
@@ -51,3 +51,4 @@
 {"type": "RECORD", "record": {"stream": "1_prefix_startwith_number", "emitted_at": 1602637990800, "data": { "id": 2, "date": "2020-09-01", "text": "hi 4"}}}
 
 {"type":"RECORD","record":{"stream":"multiple_column_names_conflicts","data":{"id":1,"User Id":"chris","user_id":42,"User id":300,"user id": 102,"UserId":101},"emitted_at":1623959926}}
+{"type":"RECORD","record":{"stream":"special_characters","data":{"id":1,"character_1":"some random special characters: ࠈൡሗ","user_id":42,"character_2":"some random special characters: žõ⥁🌞"},"emitted_at":1623959932}}
```
Comment (DoNotPanicUA) [inline airbyte-integrations/bases/base-normalization/integration_tests/resources/test_simple_streams/data_input/messages.txt:54]: If I'm not mistaken, this test will effect normalization of all destinations. Are you sure that we are safe to do it?

File: airbyte-integrations/bases/standard-destination-test/src/main/java/io/airbyte/integrations/standardtest/destination/comparator/AdvancedTestDataComparator.java [hunk 2]
```diff
@@ -91,6 +92,8 @@ protected boolean compareJsonNodes(final JsonNode expectedValue, final JsonNode
       return compareDateValues(expectedValue.asText(), actualValue.asText());
     } else if (expectedValue.isArray() && actualValue.isArray()) {
       return compareArrays(expectedValue, actualValue);
+    } else if (expectedValue.isArray() && isQuotedArray(actualValue)) {
```
Comment (DoNotPanicUA) [inline airbyte-integrations/bases/standard-destination-test/src/main/java/io/airbyte/integrations/standardtest/destination/comparator/AdvancedTestDataComparator.java:95]: this is a specific case of your destination. It should be handled inside your destination implementation.
I already fixed few destinations with the similar problem. You need to parse text values on the destination and provide actual value as true array here.

**Review by DoNotPanicUA — CHANGES_REQUESTED** [review]

File: airbyte-integrations/bases/base-normalization/integration_tests/resources/test_simple_streams/data_input/messages.txt (hunk 1 above)
Comment (sashaNeshcheret) [reply airbyte-integrations/bases/base-normalization/integration_tests/resources/test_simple_streams/data_input/messages.txt:54]: Yes, you are right. @grubberr will check if it works for all destinations and will make changes in scope of the PR if not.

File: airbyte-integrations/bases/standard-destination-test/src/main/java/io/airbyte/integrations/standardtest/destination/comparator/AdvancedTestDataComparator.java (hunk 2 above)
Comment (sashaNeshcheret) [reply airbyte-integrations/bases/standard-destination-test/src/main/java/io/airbyte/integrations/standardtest/destination/comparator/AdvancedTestDataComparator.java:95]: Thanks, will do it.

**Review by DoNotPanicUA — CHANGES_REQUESTED** [review]

File: airbyte-integrations/bases/standard-destination-test/src/main/java/io/airbyte/integrations/standardtest/destination/comparator/AdvancedTestDataComparator.java [hunk 3]
```diff
@@ -99,6 +100,10 @@ protected boolean compareJsonNodes(final JsonNode expectedValue, final JsonNode
     }
   }
 
+  protected boolean isQuotedArray(JsonNode actualValue) {
```
[the remark this answers is not in the document]
Comment (sashaNeshcheret) [reply airbyte-integrations/bases/standard-destination-test/src/main/java/io/airbyte/integrations/standardtest/destination/comparator/AdvancedTestDataComparator.java:103]: Yes, it is unused now. Thanks

**Review by DoNotPanicUA — APPROVED** [review]
❤️

**Review by grubberr — APPROVED** [review]

File: airbyte-integrations/bases/base-normalization/integration_tests/resources/redshift_normalization_migration/test_pokemon_super.sql [hunk 4]
```diff
@@ -1,6 +1,6 @@
 SELECT
     forms
 FROM
-    {{ REF('pokemon') }}
+    {{ ref('pokemon') }}
```
[the remark this answers is not in the document]
Comment (grubberr) [reply airbyte-integrations/bases/base-normalization/integration_tests/resources/redshift_normalization_migration/test_pokemon_super.sql:4]: if you know how to fix 
there is an issue for this  :)
https://github.com/airbytehq/airbyte/issues/12384

**Comment by grubberr** [comment]
I see that it's published 

```
docker rmi airbyte/normalization:0.1.78
docker run -ti --rm --entrypoint /bin/bash airbyte/normalization:0.1.78
root@b89db0535d0f:/airbyte# grep NVARCHAR normalization_code/dbt-template/macros/cross_db_utils/datatypes.sql
    NVARCHAR(max)
    NVARCHAR(max)
```

**Pull request merged**